package/base-files/files/lib/functions/system.sh

```diff
@@ -42,48 +49,67 @@ get_mac_label_json() {
 		json_select ..
 	fi
 
-	echo $macaddr
+	echo -n "$macaddr"
 }
 
 get_mac_label() {
-	local macaddr=$(get_mac_label_dt)
+	local macaddr
 
-	[ -n "$macaddr" ] || macaddr=$(get_mac_label_json)
+	[ -n "$(get_mac_label_dt)" ] &&
+		macaddr=$(get_mac_label_dt) ||
+		macaddr=$(get_mac_label_json)
 
-	echo $macaddr
+	macaddr_canonicalize "$macaddr"
 }
 
 find_mtd_chardev() {
-	local INDEX=$(find_mtd_index "$1")
-	local PREFIX=/dev/mtd
+	local mtdname="$1"
+	local PREFIX INDEX
 
-	[ -d /dev/mtd ] && PREFIX=/dev/mtd/
-	echo "${INDEX:+$PREFIX$INDEX}"
+	INDEX=$(find_mtd_index "$mtdname")
+	[ -d /dev/mtd ] && PREFIX="/dev/mtd/" || PREFIX="/dev/mtd"
+
+	echo -n "${INDEX:+$PREFIX$INDEX}"
 }
 
 mtd_get_mac_ascii() {
 	local mtdname="$1"
 	local key="$2"
-	local part
-	local mac_dirty
+	local part value prefix suffix
 
 	part=$(find_mtd_part "$mtdname")
 	if [ -z "$part" ]; then
 		echo "mtd_get_mac_ascii: partition $mtdname not found!" >&2
 		return
 	fi
 
-	mac_dirty=$(strings "$part" | sed -n 's/^'"$key"'=//p')
+	value=$(strings "$part" |
+		while read a; do
+			[[ "$a" == *"$key"* ]] && echo -n "$a" && break
+		done)
 
-	# "canonicalize" mac
-	[ -n "$mac_dirty" ] && macaddr_canonicalize "$mac_dirty"
+	value="${value#*${key}}"
+	value="${value//$operators/}"
+
+	# if quoted
+	[[ "$value" != "${value//$blockchar/}" ]] &&
+		value="${value#*${blockchar}}" && value="${value%%${blockchar}*}"
+
+	prefix="${value%%${delimiter}*}"
+	suffix="${value#*${delimiter}}"
+
+	# if no delimiter
+	[[ "$prefix" == "$suffix" ]] &&
+		prefix="${suffix:0:2}" && suffix="${suffix:2:10}"
+
+	macaddr_canonicalize "${prefix:(-2)}${suffix//$delimiter/}"
 }
 
 mtd_get_mac_text() {
-	local mtdname=$1
+	local mtdname="$1"
```

Comment:
Quoting on assignment is redundant (I know it's being done elsewhere in the file, but I'd rather fix those other cases)

Reply:
most examples I read have variables quoted, and some say its good practice

the only context for no quotes being:

> In contexts where we are absolutely sure that there are no word splitting or globbing issues or we do want word splitting and globbing"

I know these functions dont expect a space or wildcard `*` but if some random character is thrown in we don't want it to be interpreted as an operator and do something unexpected before returning